plugins/GeoIp2/GeoIP2AutoUpdater.php

```diff
@@ -798,4 +800,31 @@ protected function fetchUrl($url)
     {
         return Http::fetchRemoteFile($url);
     }
+
+    /**
+     * Updates the DB-IP URL option value so that users see
+     * the updated link in the "Download URL" field on the plugin page
+     * instead of the one that was set when Matomo was installed months
+     * or even years ago.
+     *
+     * @param  string  $option The option to check and update: either
+     * self::LOC_URL_OPTION_NAME or self::ISP_URL_OPTION_NAME
+     *
+     * @param  string  $url  The URL retrieved from the options
+     */
+    protected function updateDbIpUrlOption(string $option, string $url): void
+    {
+        $url = trim($url);
+
+        if (
+            self::isDbIpUrl($url)
+            && ($option === self::LOC_URL_OPTION_NAME || $option === self::ISP_URL_OPTION_NAME)
+        ) {
+            $latestUrl = $this->getDbIpUrlWithLatestDate($url);
+
+            if($url !== $latestUrl) {
+                Option::set($option, $latestUrl);
+            }
+        }
+    }
```

Comment:
Guess we could remove the second parameter and read the option directly in the method. Doesn't hurt to do that again, as the value will be cached once it was read. So maybe something like:

```php
    protected function updateDbIpUrlOption(string $option): void
    {
        if ($option !== self::LOC_URL_OPTION_NAME && $option !== self::ISP_URL_OPTION_NAME) {
            return;
        }
    
        $url = trim(Option::get($option));

        if (self::isDbIpUrl($url)) {
            $latestUrl = $this->getDbIpUrlWithLatestDate($url);

            if($url !== $latestUrl) {
                Option::set($option, $latestUrl);
            }
        }
    }
```

Reply:
Thanks @sgiehl, makes sense to get rid of the second parameter if option values are cached.